### PR TITLE
add [PUB] likes badge

### LINKED ISSUE #7915: add [PUB] likes badge
PUB Likes

:clipboard: **Description**

- This badge is for `Dart/Flutter` packages/plugins.
- This badge shows the pub likes for packages/plugins.

:link: **Data**

- There is a public API for this data i.e `https://pub.dev/api/packages/<packageName>/score`.
- This API doesn't require an API key and returns a JSON object.

:microphone: **Motivation**

- This badge is a measure of how many developers have liked a package.
- This provides a raw measure of the overall sentiment of a package from peer developers.

### PR DESCRIPTION
**Pull request by G1Joshi**
- Add a badge for `PUB` to show the likes for `Dart/Flutter` packages and plugins.
- Fix #7915.

### CODE REVIEW

**Comment by G1Joshi** [comment]
This PR depends on #7921.
All tests will pass after merging the above-mentioned PR.

File: services/pub/pub-likes.service.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    },
+  ]
+
+  static defaultBadgeData = { label: 'likes' }
+
+  static render({ likeCount, packageName }) {
+    return {
+      label: 'likes',
+      message: `${likeCount}`,
+      color: 'brightgreen',
```
Comment (chris48s) [inline services/pub/pub-likes.service.js:39]: If we're just going to hard-code a color (i.e: we're not applying color based on the value - which is probably correct for this badge) this should be blue for informational.

File: services/pub/pub-likes.service.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const documentation = `<p>A measure of how many developers have liked a package. This provides a raw measure of the overall sentiment of a package from peer developers.</p>`
+
+const keywords = ['dart', 'flutter']
+
+const schema = Joi.object({
+  likeCount: Joi.number().min(0).required(),
```
Comment (chris48s) [inline services/pub/pub-likes.service.js:9]: I think both this and the valid case service test can use `nonNegativeInteger`?
https://github.com/badges/shields/blob/f219e3cd193426807a5145d4992ebccebc40ec6a/services/validators.js#L11

File: services/pub/pub-likes.service.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  ]
+
+  static defaultBadgeData = { label: 'likes' }
+
+  static render({ likeCount, packageName }) {
+    return {
+      label: 'likes',
+      message: `${likeCount}`,
+      color: 'brightgreen',
+      link: `https://pub.dev/packages/${packageName}`,
```
Comment (chris48s) [inline services/pub/pub-likes.service.js:40]: Links should be off by default for non-social badges
https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines

File: services/pub/pub-likes.service.js (hunk 2 above)
Comment (G1Joshi) [reply services/pub/pub-likes.service.js]: I have added the nonNegativeInteger for like counts.

**Comment by calebcartwright** [comment]
Could you rebase when you get a chance to pull in the test fixes

**Comment by G1Joshi** [comment]
> Could you rebase when you get a chance to pull in the test fixes

Done :)

**Review by chris48s — APPROVED** [review]

**Pull request merged**